apps/web/app/environments/[environmentId]/surveys/PreviewSurvey.tsx

```diff
@@ -24,6 +26,68 @@ interface PreviewSurveyProps {
   environment: TEnvironment;
 }
 
+// Extracted modal content to prevent duplication
+const PreviewModalContent = ({
+  handleStopCountdown,
+  activeQuestionId,
+  lastActiveQuestionId,
+  thankYouCard,
+  questions,
+  brandColor,
+  gotoNextQuestion,
+  showFormbricksSignature,
+  progress,
+}) => (
+  <div>
+    <div
+      onClick={() => handleStopCountdown()}
+      onMouseOver={() => handleStopCountdown()}
+      className="px-4 py-6 sm:p-6">
+      <QuestionRenderer
+        activeQuestionId={activeQuestionId}
+        lastActiveQuestionId={lastActiveQuestionId}
+        questions={questions}
+        brandColor={brandColor}
+        thankYouCard={thankYouCard}
+        gotoNextQuestion={gotoNextQuestion}
+      />
+      {showFormbricksSignature && <FormbricksSignature />}
+    </div>
+    <Progress progress={progress} brandColor={brandColor} />
+  </div>
+);
+
+const QuestionRenderer = ({ activeQuestionId, lastActiveQuestionId, questions, brandColor, thankYouCard, gotoNextQuestion }) => {
```

Comment:
Extracted question rendering logic into a separate component as it was being used multiple times